---
title: "Marques et al 2021 WT1bOE ATACseq"
output:
  pdf_document:
    toc: yes
    fig_width: 10
    fig_height: 8
classoption: landscape
---

# ATAC seq workflow

## Preprocessing and analysis overview

The data has been analyzed in the following ways before bringing it to R:

1)  FASTQC
2)  Trimming using Trimmomatic
3)  Alignment using bowtie2
4)  Remove mitochondria and Fluorescent Proteins
5)  Remove duplicates using Picard
6)  Check peak distribution
7)  Downsample the samples to lowest reads
8)  Run genrich to call peaks

in R (this notebook)

8)  Run DiffBind for differential peaks
9)  Run ChipSeeker for annotation
10) Run Clusterprofiler for pathway enrichment

# Analysis

## Clear memory and set working directory

```{r, setup, include=FALSE}
# chunk option defaults
knitr::opts_chunk$set(fig.height=9, fig.width=9)
knitr::opts_chunk$set(dpi = 300)
knitr::opts_knit$set(root.dir = '/home/prateek/Mercader_Lab/Ines_Data/wt1oe_ATAC_Seq/final')

```

```{r}
rm(list = ls())
gc()
setwd("/home/prateek/Mercader_Lab/Ines_Data/wt1oe_ATAC_Seq/final")
getwd()
```

```{r}
library(dplyr)
library(DiffBind)
library(GenomicFeatures)
library(ChIPseeker)
library(rstatix)
library(ggpubr)
library(RColorBrewer)
library("viridis")
library(ReactomePA)
library(clusterProfiler)
library(org.Mm.eg.db)
```

# Diffbound analysis

## Read Peak files

```{r}

#make samplesheet to read the peak files
samplesheet_df <- openxlsx::read.xlsx("DiffBind_Samplesheet_generich_downsampled.xlsx")

samplesheet_df$bamReads <- paste0("only_unique_downsampled/",samplesheet_df$bamReads)
samplesheet_df$Peaks <- paste0("Generich_downsampled/", samplesheet_df$Peaks)
(samplesheet_df)
```

## Create DBA object for Diffbound

```{r}
wt1oe_db <- dba(sampleSheet = samplesheet_df)
wt1oe_db
```

## QC of samples

```{r}
plot(wt1oe_db)
```

# Count the peaks in samples

(it counts a peak if the peak is present in more than one sample)

```{r}
wt1oe_db <- dba.count(wt1oe_db, bParallel = TRUE)
```

```{r}
plot(wt1oe_db)
```

```{r}
wt1oe_db
```

## Set the FDR and pvalue

```{r}
wt1oe_db$config$th = 0.05
```

## Data Normalization for the analysis

Using "DeSeq2" and "Background" normalization as it is recommended for ATACSeq (see Diffbind vignette)

```{r}
wt1oe_db <- dba.normalize(wt1oe_db, method=DBA_ALL_METHODS,normalize=DBA_NORM_NATIVE, background=TRUE)
wt1oe_db <- dba.analyze(wt1oe_db, method=DBA_ALL_METHODS)
dba.show(wt1oe_db,bContrasts=TRUE)
```

```{r}
db_deseq2 <- dba.report(wt1oe_db,method=DBA_DESEQ2, bDB=TRUE, bGain=TRUE, bLoss=TRUE,bUsePval = T, bNormalized = T)
db_deseq2

```

```{r}
plot(wt1oe_db, contrast=1, bUsePval = T)
```

Write the differential peaks in csv file

```{r}
getwd()
dba.report(wt1oe_db,method=DBA_DESEQ2, bCalledDetail = T,file = "./results/DBA_wt1oe_downsampled_3_samples_generich_deseq_norm",
           bGain=TRUE, bLoss=TRUE,bUsePval = T, bNormalized = T)
```

```{r}
db_deseq2_df <- read.csv("./results/DBA_wt1oe_downsampled_3_samples_generich_deseq_norm.csv")
head(db_deseq2_df)
?dba.report
```

```{r}

diff_peaks_pval_norm <- as.data.frame(db_deseq2_df)
head(diff_peaks_pval_norm)
write.csv(diff_peaks_pval_norm, "./results/diff_peaks_downsample_generich_3_samples_pval_norm.csv")
write.table(diff_peaks_pval_norm, "./results/diff_peaks_downsample_generich_3_samples_pval_norm.bed",
            quote=F, sep="\t", row.names=F, col.names=F)
```

## Plot PCA and Boxplot for normalized data

```{r}
dba.plotPCA(wt1oe_db,DBA_CONDITION,label=DBA_ID)
pvals <- dba.plotBox(wt1oe_db,bUsePval = T, notch= FALSE)
```

# Annotation of Peaks

## Get the zebrafish genome data from Ensembl (a TxDB object that can be used for ChiPSeeker)

```{r}
TxDBDR102 <- makeTxDbFromEnsembl(organism="Danio rerio",
                    release=102,
                    circ_seqs=NULL,
                    server="ensembldb.ensembl.org",
                    username="anonymous", password=NULL, port=0L,
                    tx_attrib=NULL)
TxDBDR102
```

```{r}
saveDb(TxDBDR102,file = "Danio_rerio_annotationdbi_ensbl_v102.sqlite")
# txdb <- TxDBDR102
txdb <- loadDb("Danio_rerio_annotationdbi_ensbl_v102.sqlite")


```

## Make .bed files from narrowpeak files (in bash)

```{bash}
inputFiles_R1="./trim_paired/input_files_R1.txt"
array_R1=($(<$inputFiles_R1))


for SAMPLE in {0..5} 
do
#set input file 1 to "FL1", input file 2 to "FL2"
FL1=${array_R1[SAMPLE]}
#echo ${FL1}
FL1_trimmed=${FL1%_L1*}
echo ${FL1_trimmed}

cut -f 1-6 ./Generich_downsampled/${FL1_trimmed}_downsampled.narrowPeak > ./Generich_downsampled/${FL1_trimmed}_downsampled_narrowpeak.bed

done

```

## Read .bed files

```{r}
peaks_files <- list.files(path = "./Generich_downsampled/", pattern = "*narrowpeak.bed")
peaks_files
class(peaks_files)
remove_files <- peaks_files[grep(pattern = "bdg", peaks_files)]
remove_files
peaks_files <-  peaks_files[!(peaks_files %in% remove_files)]
peaks_files

sample_names <- gsub("_downsampled_narrowpeak.bed","", peaks_files)
sample_names

```

# Annotation of peaks using ChipSeeker

## Read narrowpeak bed files

```{r}
peakfiles_granges <-list.files(path = "./Generich_downsampled/", pattern = "narrowpeak.bed")

peakfiles_granges <- peakfiles_granges[!peakfiles_granges %in% remove_files]
peakfiles_granges

peakfiles_granges <- paste0("./Generich_downsampled/",peakfiles_granges)
peakfiles_granges


read_peaks <- readPeakFile(peakfiles_granges[1])
read_peaks

peakfiles_granges_names <- peakfiles_granges
peakfiles_granges_names <- as.list(peakfiles_granges_names)
names(peakfiles_granges_names) <- sample_names
peakfiles_granges_names

```

## Annotate the peaks

```{r}
peakAnnoList <- lapply(peakfiles_granges_names, annotatePeak, tssRegion=c(-1000, 1000), TxDb=txdb, verbose=FALSE)
```

## Check the genomic regions assigned to the peaks (promoter, exon, introns, etc)

```{r}
plotAnnoBar(peakAnnoList)
dev.copy(
svg,
file = paste0("./results/Peak_Annotation.svg"),
width = 10,
height = 8
)
dev.off ()
```

## Check the distance to Transcription Start Site (TSS)

```{r}
plotDistToTSS(peakAnnoList)
dev.copy(
svg,
file = paste0("./results/Distance_to_TSS.svg"),
width = 10,
height = 8
)
dev.off ()

```

## Compare if the percentage and absolute number of peaks differ in different in Wt1bOE vs Ctrl

```{r}
# Convert peak stats to dataframe
peakAnnoList %>% head
length(peakAnnoList)
annotstat_df <- data.frame()
for (i in 1:length(peakAnnoList)) {
    tmp_df <- as.data.frame(peakAnnoList[[i]]@annoStat)
    tmp_df$sample <- names(peakAnnoList)[i]
    annotstat_df <- bind_rows(annotstat_df, tmp_df)
}
(annotstat_df)
```

```{r}
annotstat_df$condition <- NA
annotstat_df$condition[grep(pattern = "URV",annotstat_df$sample)] <- "Control"
annotstat_df$condition[grep(pattern = "wt1b",annotstat_df$sample)] <- "WT1bOE"
annotstat_df
```

```{r}

```

```{r}
annotstat_df
#some samples do not have 1st exon,. this needs to be added manually to help conduct ttest

add_df_rows_exon <- data.frame(Feature= c("1st Exon","1st Exon"), Frequency= c(0,0), 
                               sample= c("wt1bOE_2", "wt1bOE_3"), condition= c("WT1bOE", "WT1bOE"))
annotstat_df <- rbind(annotstat_df, add_df_rows_exon)

annotstat.ttest <- annotstat_df %>%
  group_by(Feature) %>% 
  t_test(Frequency ~ condition) %>% 
  add_significance()
annotstat.ttest
```

```{r}
annotstat.ttest <- annotstat.ttest %>% add_xy_position(x = "Feature")
annotstat.ttest

q <- ggboxplot(
  annotstat_df, x = "Feature", y = "Frequency",
  fill = "condition", legend= "right") +
  theme(axis.text.x = element_text(angle = 30, hjust = 1))+
  stat_pvalue_manual(annotstat.ttest, label = "p")+
  labs(x= "Feature Type", y= "Mean Percentage", title = "Various Genomic features in Control vs  WT1bOE")

print(q)


```

Write the annotated peaks for each sample

```{r}

for(i in 1:length(peakfiles_granges_names)){
# for(i in 1:1){
    tmp_df <- as.data.frame(peakAnnoList[[i]]@anno)
    write.csv(tmp_df, file=paste0("./annotated_peaks/", 
                                  sample_names[i], "_annotations",".csv"),
              row.names=F, col.names=T)
}


```

## Check the absolute counts of annotated peaks

```{r}
read_peaks <- function(filename){
  print(filename)
  df <- read.csv(paste0("./annotated_peaks/", filename))
  df$sample_name <- gsub("_annotations.csv","", filename)
  if((grepl(pattern = "URV", filename))){df$condition <-"Control"
  } else if(grepl(pattern = "wt1", filename)) {df$condition <-"WT1bOE"}
  
  df$annotation <- gsub("\\(EN.*","", df$annotation)
  df
  
}
```

```{r}
peaks_files <- list.files(path = "./annotated_peaks/", pattern = "csv")

```

```{r}
combine_peaks_df<- lapply(X = peaks_files, FUN = read_peaks) %>% bind_rows()
head(combine_peaks_df)

combine_peaks_df_counts <- combine_peaks_df %>% group_by(condition, sample_name, annotation) %>%
  summarise(counts= n()) %>% group_by(condition, annotation) %>% 
  mutate(sd=sd(counts), mean=mean(counts)) %>% 
  arrange(desc(counts))
combine_peaks_df_counts$annotation <- factor(combine_peaks_df_counts$annotation, 
                                             levels = unique(combine_peaks_df_counts$annotation))

head(combine_peaks_df_counts)
```

write csv with total peak counts

```{r}
write.csv(combine_peaks_df_counts, "./results/total_peak_counts.csv")

```

```{r}
q <- ggplot(data= combine_peaks_df_counts, aes(x=annotation, y= mean, fill= condition))+
  geom_bar(stat = "identity", position = "dodge")+ 
    geom_errorbar(aes(ymin= mean-sd, ymax= mean+sd),width=.2,
                 position=position_dodge(.9))+
  theme(axis.text.x = element_text(angle= 45, hjust=1))
print(q)
dev.copy(
svg,
file = paste0("./results/Total_peak_counts_annotations.svg"),
width = 10,
height = 8
)
dev.off ()
```

## Annotation of the differential peaks and representation in graphs (promoters, exons, introns, etc)

```{r}
peakfiles_granges_diff <- "./results/diff_peaks_downsample_generich_3_samples_pval_norm.bed"
read_peaks_diff <- readPeakFile(peakfiles_granges_diff)
read_peaks_diff

peakAnno_diff <- annotatePeak(read_peaks_diff, TxDb = txdb)
peakAnno_diff
#plot the pie chart with all the info on peak distribution
plotAnnoPie(peakAnno_diff)
vennpie(peakAnno_diff)
```

```{r}

plotAnnoPie(peakAnno_diff, col = rainbow(10, alpha = 0.7, s = 0.8),border= "white")
# display.brewer.all()
dev.copy(
svg,
file = paste0("./results/Piechart_rainbow.svg"),
width = 10,
height = 8
)
dev.off ()




```

```{r}
vennpie(peakAnno_diff)
```

## Correct the names of the annotated dataframe of peaks

```{r}
peakAnno_diff@anno
peakAnno_diff_df <- as.data.frame(peakAnno_diff@anno)
head(peakAnno_diff_df)
colnames(peakAnno_diff_df)[7:12] <- colnames(diff_peaks_pval_norm)[5:10]
peakAnno_diff_df %>% head()
peakAnno_diff_df %>% names
as.data.frame(db_deseq2_df) %>% head
diff_peaks_pval_norm %>% head()

```

## Convert genes names to symbol and entrezid

```{r}
Genes=peakAnno_diff_df$geneId

ensmbl_to_symbol_danio <- function(x){
  require(biomaRt)
  mart <- useMart(biomart = "ensembl", dataset = "drerio_gene_ensembl")
  # query biomart
  results <- getBM(attributes = c("ensembl_gene_id", "zfin_id_symbol",
                                  "entrezgene_id", "description"),
                                  filters = "ensembl_gene_id", values = x,
                                  mart = mart)
  results
                                      }
zebrafish_genes <- ensmbl_to_symbol_danio(Genes)
head(zebrafish_genes)
```

```{r}
head(peakAnno_diff_df)
peakAnno_diff_df_entrez <- merge(peakAnno_diff_df, zebrafish_genes, by.x= "geneId", by.y = "ensembl_gene_id", all.x=T)
(peakAnno_diff_df_entrez) %>% filter(geneId== "ENSDARG00000104380")

```

```{r}
dim(peakAnno_diff_df_entrez)
```

```{r}
peakAnno_diff_df_entrez <- peakAnno_diff_df_entrez %>% arrange(p.value)
head(peakAnno_diff_df_entrez)
```

Write the csv for annotated differential peaks

```{r}
write.csv(peakAnno_diff_df_entrez, file= "./results/differential_peaks_with_annotations.csv",row.names=F,col.names=T)
```

# Pathway analysis

## Convert zebrafish genes to mouse genes and merge to original dataframe

```{r}
zgGenes <- peakAnno_diff_df_entrez$geneId
# Basic function to convert zebrafish to human gene names


convertDanioGeneList_Mouse <- function(x){
  require("biomaRt")
  mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")
  danio = useMart("ensembl", dataset = "drerio_gene_ensembl")
  genesV2 = getLDS(attributes = c("ensembl_gene_id", "zfin_id_symbol"), 
                   filters = "ensembl_gene_id", values = x , 
                   mart = danio, attributesL = c("mgi_symbol", "ensembl_gene_id","entrezgene_id"), 
                                martL = mouse, uniqueRows=T)
  
  colnames(genesV2)[colnames(genesV2)== "Gene.stable.ID"] <- "EnsmblID_Zebrafish"
  colnames(genesV2)[colnames(genesV2)== "NCBI.gene..formerly.Entrezgene..ID"] <- "EntrezID_Mouse"
  colnames(genesV2)[colnames(genesV2)== "Gene.stable.ID.1"] <- "EnsmblID_Mouse"
  
  # Print the first 6 genes found to the screen
  return(genesV2)
}

Mouse_Genes <- convertDanioGeneList_Mouse(zgGenes)
Mouse_Genes %>% head
```

```{r}
peakAnno_diff_mouse <- merge(peakAnno_diff_df_entrez, Mouse_Genes, by.x= "geneId", by.y = "EnsmblID_Zebrafish", all.x= TRUE)
head(peakAnno_diff_mouse)
dim(peakAnno_diff_mouse)
```

```{r}
write.csv(peakAnno_diff_mouse, "./results/mouse_differential_peaks_annotations.csv")
```

## Perfroming enrichment analysis on differential peaks/genes.

```{r, fig.width = 12, fig.height = 12}
compKEGG_diff <- enrichKEGG(peakAnno_diff_mouse$EntrezID_Mouse, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",organism = "mmu")
dotplot(compKEGG_diff, showCategory = 25, title = "KEGG Pathway Enrichment Analysis DEPs")
```

```{r, fig.width = 12, fig.height = 12}
compGO_MF_diff <- enrichGO(gene = peakAnno_diff_mouse$EntrezID_Mouse, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",OrgDb = "org.Mm.eg.db", ont = "MF",readable = TRUE)
  
dotplot(compGO_MF_diff, showCategory = 15, title = "GO Pathway Enrichment Analysis \n Molecular Functions for DEPs", 
        font.size = 12)

compGO_BP_diff <- enrichGO(gene = peakAnno_diff_mouse$EntrezID_Mouse, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",OrgDb = "org.Mm.eg.db", ont = "BP",readable = TRUE)
  
dotplot(compGO_BP_diff, showCategory = 15, title = "GO Pathway Enrichment Analysis \n Biological Pathways for DEPs", 
        font.size = 12)

compGO_CC_diff <- enrichGO(gene = peakAnno_diff_mouse$EntrezID_Mouse, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",OrgDb = "org.Mm.eg.db", ont = "CC",readable = TRUE)
  
dotplot(compGO_CC_diff, showCategory = 15, title = "GO Pathway Enrichment Analysis \n Cellular components for DEPs", 
        font.size = 12)
```

```{r, fig.width = 12, fig.height = 12, fig.caption="Reactome Pathway Enrichment Analysis on Differential Peaks"}
reactome_pathways_diff <- enrichPathway(peakAnno_diff_mouse$EntrezID_Mouse, organism = "mouse", readable = T,
                                        pvalueCutoff = 0.05, qvalueCutoff = 0.2)
head(summary(reactome_pathways_diff))
dotplot(reactome_pathways_diff, showCategory = 25, title = "Reactome Pathway Enrichment Analysis", 
        font.size = 12)
```

Write the table for diffential pathways

```{r}
compGO_CC_diff_df <- as.data.frame(compGO_CC_diff)


compGO_CC_diff_df$GeneRatio_decimal <- compGO_CC_diff_df$GeneRatio
compGO_CC_diff_df$GeneRatio_decimal <- sapply(compGO_CC_diff_df$GeneRatio_decimal, 
                                              function(x) (eval(parse(text = as.character(x)))))
compGO_CC_diff_df$BgRatio_decimal <- compGO_CC_diff_df$BgRatio
compGO_CC_diff_df$BgRatio_decimal <- sapply(compGO_CC_diff_df$BgRatio_decimal, 
                                            function(x) (eval(parse(text = as.character(x)))))
compGO_CC_diff_df <- compGO_CC_diff_df %>% tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>%
  arrange(desc(GeneRatio_decimal))
compGO_CC_diff_df %>% head

write.csv(compGO_CC_diff_df, "results/differential_pathways_GO_Cell_components_norm.csv")
```

```{r}
compGO_MF_diff_df <- as.data.frame(compGO_MF_diff)


compGO_MF_diff_df$GeneRatio_decimal <- compGO_MF_diff_df$GeneRatio
compGO_MF_diff_df$GeneRatio_decimal <- sapply(compGO_MF_diff_df$GeneRatio_decimal, 
                                              function(x) (eval(parse(text = as.character(x)))))
compGO_MF_diff_df$BgRatio_decimal <- compGO_MF_diff_df$BgRatio
compGO_MF_diff_df$BgRatio_decimal <- sapply(compGO_MF_diff_df$BgRatio_decimal, 
                                            function(x) (eval(parse(text = as.character(x)))))
compGO_MF_diff_df <- compGO_MF_diff_df %>% tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>%
  arrange(desc(GeneRatio_decimal))
compGO_MF_diff_df %>% head

write.csv(compGO_MF_diff_df, "results/differential_pathways_GO_Molecular_Functions_norm.csv")

```

```{r}
compGO_BP_diff_df <- as.data.frame(compGO_BP_diff)


compGO_BP_diff_df$GeneRatio_decimal <- compGO_BP_diff_df$GeneRatio
compGO_BP_diff_df$GeneRatio_decimal <- sapply(compGO_BP_diff_df$GeneRatio_decimal, 
                                              function(x) (eval(parse(text = as.character(x)))))
compGO_BP_diff_df$BgRatio_decimal <- compGO_BP_diff_df$BgRatio
compGO_BP_diff_df$BgRatio_decimal <- sapply(compGO_BP_diff_df$BgRatio_decimal, 
                                            function(x) (eval(parse(text = as.character(x)))))
compGO_BP_diff_df <- compGO_BP_diff_df %>% tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>%
  arrange(desc(GeneRatio_decimal))
compGO_BP_diff_df %>% head

write.csv(compGO_BP_diff_df, "results/differential_pathways_GO_Biological_Pathways_norm.csv")

```

```{r}
compGO_MF_diff_df_plot <- as.data.frame(compGO_MF_diff)
compGO_MF_diff_df_plot$GeneRatio_decimal <- compGO_MF_diff_df_plot$GeneRatio
compGO_MF_diff_df_plot$GeneRatio_decimal <- sapply(compGO_MF_diff_df_plot$GeneRatio_decimal, 
                                                   function(x) (eval(parse(text = as.character(x)))))
compGO_MF_diff_df_plot <- compGO_MF_diff_df_plot %>% arrange(desc(GeneRatio_decimal))
head(compGO_MF_diff_df_plot, 25)
compGO_MF_diff_df_plot[1:25,]
ggplot(compGO_MF_diff_df_plot[1:25,], # you can replace the numbers to the 
       # row number of pathway of your interest
             aes(x = GeneRatio_decimal, y = reorder(Description, GeneRatio_decimal))) + 
             geom_point(aes(size = Count, color = Count)) +
             theme_bw(base_size = 14) +
            scale_size_area()+
             scale_colour_gradient( low="blue", high = "red", limits= c(10,70)) +
              theme(axis.text.y = element_text(size=9.5))+
            scale_x_continuous(breaks=seq(0.02, 0.04, 0.01))+
             ylab(NULL) +
             ggtitle("GO Molecular Functions \n pathway enrichment")

dev.copy(
svg,
file = paste0("./results/GO_MF_pathways.svg"),
width = 10,
height = 8
)
dev.off ()
```

```{r}
compGO_CC_diff_df_plot <- as.data.frame(compGO_CC_diff)
compGO_CC_diff_df_plot$GeneRatio_decimal <- compGO_CC_diff_df_plot$GeneRatio
compGO_CC_diff_df_plot$GeneRatio_decimal <- sapply(compGO_CC_diff_df_plot$GeneRatio_decimal, 
                                                   function(x) (eval(parse(text = as.character(x)))))
compGO_CC_diff_df_plot <- compGO_CC_diff_df_plot %>% arrange(desc(GeneRatio_decimal))
head(compGO_CC_diff_df_plot, 25)
compGO_CC_diff_df_plot[1:25,]
ggplot(compGO_CC_diff_df_plot[1:25,], # you can replace the numbers to the row number of pathway of your interest
             aes(x = GeneRatio_decimal, y = reorder(Description, GeneRatio_decimal))) + 
             geom_point(aes(size = Count, color = Count)) +
             scale_size_area()+
             theme_bw(base_size = 14) +
             scale_colour_gradient( low="blue", high = "red", limits= c(10,70)) +
              theme(axis.text.y = element_text(size=9.5))+
             ylab(NULL) +
             ggtitle("GO Cellular Component pathways enrichment")

dev.copy(
svg,
file = paste0("./results/GO_CC_pathways.svg"),
width = 10,
height = 8
)
dev.off ()
```

```{r}
compGO_BP_diff_df_plot <- as.data.frame(compGO_BP_diff)
compGO_BP_diff_df_plot$GeneRatio_decimal <- compGO_BP_diff_df_plot$GeneRatio
compGO_BP_diff_df_plot$GeneRatio_decimal <- sapply(compGO_BP_diff_df_plot$GeneRatio_decimal, 
                                                   function(x) (eval(parse(text = as.character(x)))))
compGO_BP_diff_df_plot <- compGO_BP_diff_df_plot %>% arrange(desc(GeneRatio_decimal))
head(compGO_BP_diff_df_plot, 25)
compGO_BP_diff_df_plot[1:25,]
ggplot(compGO_BP_diff_df_plot[1:25,], # you can replace the numbers to the row number of pathway of your interest
             aes(x = GeneRatio_decimal, y = reorder(Description, GeneRatio_decimal))) + 
             geom_point(aes(size = Count, color = Count)) +
              scale_size_area()+
             theme_bw(base_size = 14) +
             scale_colour_gradient( low="blue", high = "red", limits= c(10,70)) +
              theme(axis.text.y = element_text(size=12))+
             ylab(NULL) +
             ggtitle("GO Biological Pathways enrichment")


dev.copy(
svg,
file = paste0("./results/GO_BP_pathways.svg"),
width = 10,
height = 8
)
dev.off ()
```

```{r}

compKEGG_diff_readable <- setReadable(compKEGG_diff, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
## The geneID column is translated to symbol
head(compKEGG_diff_readable, 3)
```

```{r}
compKEGG_diff_readable_df <- as.data.frame(compKEGG_diff_readable)


compKEGG_diff_readable_df$GeneRatio_decimal <- compKEGG_diff_readable_df$GeneRatio
compKEGG_diff_readable_df$GeneRatio_decimal <- sapply(compKEGG_diff_readable_df$GeneRatio_decimal, function(x) (eval(parse(text = as.character(x)))))
compKEGG_diff_readable_df$BgRatio_decimal <- compKEGG_diff_readable_df$BgRatio
compKEGG_diff_readable_df$BgRatio_decimal <- sapply(compKEGG_diff_readable_df$BgRatio_decimal, function(x) (eval(parse(text = as.character(x)))))
compKEGG_diff_readable_df <- compKEGG_diff_readable_df %>% tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>% arrange(desc(GeneRatio_decimal))
compKEGG_diff_readable_df %>% head

write.csv(compKEGG_diff_readable_df, "./results/differential_pathways_KEGG_Pathways.csv")
```

Writing all pathways reactome pathways in excel files-- suggested cutoff-- qvalue\<0.2

```{r}
reactome_pathways_diff_df <- as.data.frame(reactome_pathways_diff@result)


reactome_pathways_diff_df$GeneRatio_decimal <- reactome_pathways_diff_df$GeneRatio
reactome_pathways_diff_df$GeneRatio_decimal <- sapply(reactome_pathways_diff_df$GeneRatio_decimal, function(x) (eval(parse(text = as.character(x)))))
reactome_pathways_diff_df$BgRatio_decimal <- reactome_pathways_diff_df$BgRatio
reactome_pathways_diff_df$BgRatio_decimal <- sapply(reactome_pathways_diff_df$BgRatio_decimal, function(x) (eval(parse(text = as.character(x)))))
reactome_pathways_diff_df <- reactome_pathways_diff_df %>% tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>% arrange( qvalue,desc(GeneRatio_decimal))
reactome_pathways_diff_df %>% head

write.csv(reactome_pathways_diff_df, "./results/differential_pathways_Reactome_Pathways_norm.csv")
```

## Perform GO analysis on genes common in CHIPT-ATLAS and current ATACseq analysis

```{r}
chip_atac_common <- openxlsx::read.xlsx("CHIP_ATLAS/chipatlas_genrich_common.xlsx")

head(chip_atac_common)

```

```{r}
Genes <- chip_atac_common$`426.common.elements.in."chip-atlas".and."ATAC":`

symbol_to_entrez_mouse <- function(x){
  require(biomaRt)
  mart <- useMart(biomart = "ensembl", dataset = "mmusculus_gene_ensembl")
  # query biomart
  results <- getBM(attributes = c("entrezgene_id", "mgi_symbol"),
                                  filters = "mgi_symbol", values = x,
                                  mart = mart)
  results
                                    }
common_mouse_entrez <- symbol_to_entrez_mouse(Genes)
head(common_mouse_entrez)

```

```{r, fig.width = 12, fig.height = 12}
compGO_common_MF_diff <- enrichGO(gene = common_mouse_entrez$entrezgene_id, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",OrgDb = "org.Mm.eg.db", ont = "MF",readable = TRUE)
  
dotplot(compGO_common_MF_diff, showCategory = 15, 
        title = "CHIPATLAS common peaks GO Pathway Enrichment Analysis \n Molecular Functions for DARs",
        font.size = 12)

compGO_common_BP_diff <- enrichGO(gene = common_mouse_entrez$entrezgene_id, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",OrgDb = "org.Mm.eg.db", ont = "BP",readable = TRUE)
  
dotplot(compGO_common_BP_diff, showCategory = 15, 
        title = "CHIPATLAS common peaks GO Pathway Enrichment Analysis \n Biological Pathways for DARs",
        font.size = 12)

compGO_common_CC_diff <- enrichGO(gene = common_mouse_entrez$entrezgene_id, pvalueCutoff  = 0.05,
                         pAdjustMethod = "BH",OrgDb = "org.Mm.eg.db", ont = "CC",readable = TRUE)
  
dotplot(compGO_common_CC_diff, showCategory = 15, 
        title = "CHIPATLAS common peaks GO Pathway Enrichment Analysis \n Cellular components for DARs", 
        font.size = 12)


```

```{r}
compGO_common_CC_diff_df <- as.data.frame(compGO_common_CC_diff)


compGO_common_CC_diff_df$GeneRatio_decimal <- compGO_common_CC_diff_df$GeneRatio
compGO_common_CC_diff_df$GeneRatio_decimal <- sapply(compGO_common_CC_diff_df$GeneRatio_decimal, 
                                                     function(x) (eval(parse(text = as.character(x)))))
compGO_common_CC_diff_df$BgRatio_decimal <- compGO_common_CC_diff_df$BgRatio
compGO_common_CC_diff_df$BgRatio_decimal <- sapply(compGO_common_CC_diff_df$BgRatio_decimal, 
                                                   function(x) (eval(parse(text = as.character(x)))))
compGO_common_CC_diff_df <- compGO_common_CC_diff_df %>% 
  tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>% 
  arrange(desc(GeneRatio_decimal))
compGO_common_CC_diff_df %>% head

write.csv(compGO_common_CC_diff_df, 
          "results/chip_common_differential_pathways_GO_Cell_components_norm.csv")
```

```{r}
compGO_common_MF_diff_df <- as.data.frame(compGO_common_MF_diff)


compGO_common_MF_diff_df$GeneRatio_decimal <- compGO_common_MF_diff_df$GeneRatio
compGO_common_MF_diff_df$GeneRatio_decimal <- sapply(compGO_common_MF_diff_df$GeneRatio_decimal, 
                                                     function(x) (eval(parse(text = as.character(x)))))
compGO_common_MF_diff_df$BgRatio_decimal <- compGO_common_MF_diff_df$BgRatio
compGO_common_MF_diff_df$BgRatio_decimal <- sapply(compGO_common_MF_diff_df$BgRatio_decimal, 
                                                   function(x) (eval(parse(text = as.character(x)))))
compGO_common_MF_diff_df <- compGO_common_MF_diff_df %>% 
  tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>% 
  arrange(desc(GeneRatio_decimal))
compGO_common_MF_diff_df %>% head

write.csv(compGO_common_MF_diff_df, "results/chip_common_differential_pathways_GO_Molecular_Functions_norm.csv")

```

```{r}
compGO_common_BP_diff_df <- as.data.frame(compGO_common_BP_diff)


compGO_common_BP_diff_df$GeneRatio_decimal <- compGO_common_BP_diff_df$GeneRatio
compGO_common_BP_diff_df$GeneRatio_decimal <- sapply(compGO_common_BP_diff_df$GeneRatio_decimal, 
                                                     function(x) (eval(parse(text = as.character(x)))))
compGO_common_BP_diff_df$BgRatio_decimal <- compGO_common_BP_diff_df$BgRatio
compGO_common_BP_diff_df$BgRatio_decimal <- sapply(compGO_common_BP_diff_df$BgRatio_decimal, 
                                                   function(x) (eval(parse(text = as.character(x)))))
compGO_common_BP_diff_df <- compGO_common_BP_diff_df %>% 
  tidyr::separate_rows(geneID, sep = "/", convert = FALSE) %>% 
  arrange(desc(GeneRatio_decimal))
compGO_common_BP_diff_df %>% head

write.csv(compGO_common_BP_diff_df,
          "results/chip_common_differential_pathways_GO_Biological_Pathways_norm.csv")

```

```{r}
compGO_common_MF_diff_df_plot <- as.data.frame(compGO_common_MF_diff)
compGO_common_MF_diff_df_plot$GeneRatio_decimal <- compGO_common_MF_diff_df_plot$GeneRatio
compGO_common_MF_diff_df_plot$GeneRatio_decimal <- sapply(compGO_common_MF_diff_df_plot$GeneRatio_decimal, 
                                                          function(x) (eval(parse(text = as.character(x)))))
compGO_common_MF_diff_df_plot <- compGO_common_MF_diff_df_plot %>% 
  arrange(desc(GeneRatio_decimal))
head(compGO_common_MF_diff_df_plot, 25)
compGO_common_MF_diff_df_plot[1:25,]
ggplot(compGO_common_MF_diff_df_plot[1:25,], # you can replace the numbers 
       #to the row number of pathway of your interest
             aes(x = GeneRatio_decimal, y = reorder(Description, GeneRatio_decimal))) + 
             geom_point(aes(size = Count, color = Count)) +
             theme_bw(base_size = 14) +
            scale_size_area()+
             scale_colour_gradient( low="blue", high = "red", limits= c(5,70)) +
              theme(axis.text.y = element_text(size=9.5))+
            # scale_x_continuous(breaks=seq(0.02, 0.08, 0.01))+
             ylab(NULL) +
             ggtitle("GO Molecular Functions \n pathway enrichment")

dev.copy(
svg,
file = paste0("./results/chip_common_GO_MF_pathways.svg"),
width = 10,
height = 8
)
dev.off ()
```

```{r}
compGO_common_CC_diff_df_plot <- as.data.frame(compGO_common_CC_diff)
compGO_common_CC_diff_df_plot$GeneRatio_decimal <- compGO_common_CC_diff_df_plot$GeneRatio
compGO_common_CC_diff_df_plot$GeneRatio_decimal <- sapply(compGO_common_CC_diff_df_plot$GeneRatio_decimal,
                                                          function(x) (eval(parse(text = as.character(x)))))
compGO_common_CC_diff_df_plot <- compGO_common_CC_diff_df_plot %>% arrange(desc(GeneRatio_decimal))
head(compGO_common_CC_diff_df_plot, 25)
compGO_common_CC_diff_df_plot[1:25,]
ggplot(compGO_common_CC_diff_df_plot[1:25,], # you can replace the numbers to 
       #the row number of pathway of your interest
             aes(x = GeneRatio_decimal, y = reorder(Description, GeneRatio_decimal))) + 
             geom_point(aes(size = Count, color = Count)) +
             scale_size_area()+
             theme_bw(base_size = 14) +
             scale_colour_gradient( low="blue", high = "red", limits= c(5,70)) +
              theme(axis.text.y = element_text(size=9.5))+
      # scale_x_continuous(breaks=seq(0.02, 0.08, 0.01))+
             ylab(NULL) +
             ggtitle("GO Cellular Component pathways enrichment")

dev.copy(
svg,
file = paste0("./results/chip_common_GO_CC_pathways.svg"),
width = 10,
height = 8
)
dev.off ()
```

```{r}
compGO_common_BP_diff_df_plot <- as.data.frame(compGO_common_BP_diff)
compGO_common_BP_diff_df_plot$GeneRatio_decimal <- compGO_common_BP_diff_df_plot$GeneRatio
compGO_common_BP_diff_df_plot$GeneRatio_decimal <- 
  sapply(compGO_common_BP_diff_df_plot$GeneRatio_decimal, 
         function(x) (eval(parse(text = as.character(x)))))
compGO_common_BP_diff_df_plot <- compGO_common_BP_diff_df_plot %>% 
  arrange(desc(GeneRatio_decimal))

head(compGO_common_BP_diff_df_plot, 25)
compGO_common_BP_diff_df_plot[1:25,]
ggplot(compGO_common_BP_diff_df_plot[1:25,], # you can replace the numbers 
       # to the row number of pathway of your interest
             aes(x = GeneRatio_decimal, y = reorder(Description, GeneRatio_decimal))) + 
             geom_point(aes(size = Count, color = Count)) +
              scale_size_area()+
             theme_bw(base_size = 14) +
             scale_colour_gradient( low="blue", high = "red", limits= c(5,70)) +
              theme(axis.text.y = element_text(size=12))+
    # scale_x_continuous(breaks=seq(0.02, 0.08, 0.01))+
             ylab(NULL) +
             ggtitle("GO Biological Pathways enrichment")


dev.copy(
svg,
file = paste0("./results/chip_common_GO_BP_pathways.svg"),
width = 12,
height = 8
)
dev.off ()
```

# Save RData and end of program

```{r}
save.image("Marques_et_al_2021_ATACseq.RData")
```

```{r}
sessionInfo()
```
